examples/observability_sysdig/providers.tf

```diff
@@ -1,3 +1,15 @@
 provider "ibm" {
   ibmcloud_api_key = var.ibmcloud_api_key
 }
+
+provider "logdna" {
+  alias      = "at"
+  servicekey = module.test_observability_instance_creation.activity_tracker_resource_key != null ? module.test_observability_instance_creation.activity_tracker_resource_key : ""
+  url        = local.at_endpoint
+}
+
+provider "logdna" {
+  alias      = "ld"
+  servicekey = module.test_observability_instance_creation.logdna_resource_key != null ? module.test_observability_instance_creation.logdna_resource_key : ""
```

Comment:
What happens when the logic passes a blank string for servicekey value (aka if a resource key is not created if consumer chooses not to provision logdna)? Does the module give any errors? I think the provider block is still required right? But not used?

Reply:
There are no errors. The sysdig example (no logdna) plans and applies just fine. I assume the provider is evaluated just in time and only if the logdna_archive resource using it is in the dependancy tree.
I have checked manually that following a plan using archive that the archive option is set by viewing the logdna/at in the UI observability section and making sure archiving is enabled.
From memory, I think I tried null but that failed a pre-commit type check. So I made it a blank